### PR TITLE
Check and return error if failed adding to cache manager

### PR DESCRIPTION
**Pull request by rafiw**
Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/678/ for details (Mellanox internal link).

**Comment by ilansmith** [comment]
The fix looks fine.
Please change the commit message according to conventions (currently failing Jenkins).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/681/ for details (Mellanox internal link).

File: src/vma/netlink/netlink_compatibility.cpp [hunk 1]
```diff
@@ -176,6 +176,11 @@ nl_cache_mngr* nl_cache_mngr_compatible_alloc(nl_socket_handle* handle, int prot
 
 int nl_cache_mngr_compatible_add(struct nl_cache_mngr*	mngr, const char* name, change_func_t cb, void*	, struct nl_cache** result){
 	*result = nl_cache_mngr_add(mngr, name, cb);
+	if (*result == NULL) {
+		nl_logerr("Fail adding to cache manager, error=%d %s\n",
+			nl_get_errno(), nl_geterror());
+		return -1;
```
Comment (OphirMunk) [inline src/vma/netlink/netlink_compatibility.cpp]: Please note that in netlink library errors are positive number. Please explain why you return -1
https://sourcecodebrowser.com/libnl3/3.2.7/errno_8h_source.html
Maybe should return NLE_FAILURE (however NLE_FAILURE is not defined in VMA)
Please investigate more.
```
00019 #define NLE_SUCCESS         0
00020 #define NLE_FAILURE         1
00021 #define NLE_INTR            2
00022 #define NLE_BAD_SOCK        3
00023 #define NLE_AGAIN           4
00024 #define NLE_NOMEM           5
00025 #define NLE_EXIST           6
00026 #define NLE_INVAL           7
00027 #define NLE_RANGE           8
00028 #define NLE_MSGSIZE         9
00029 #define NLE_OPNOTSUPP              10
00030 #define NLE_AF_NOSUPPORT    11
00031 #define NLE_OBJ_NOTFOUND    12
00032 #define NLE_NOATTR          13
00033 #define NLE_MISSING_ATTR    14
00034 #define NLE_AF_MISMATCH            15
00035 #define NLE_SEQ_MISMATCH    16
00036 #define NLE_MSG_OVERFLOW    17
00037 #define NLE_MSG_TRUNC              18
00038 #define NLE_NOADDR          19
00039 #define NLE_SRCRT_NOSUPPORT 20
00040 #define NLE_MSG_TOOSHORT    21
00041 #define NLE_MSGTYPE_NOSUPPORT      22
00042 #define NLE_OBJ_MISMATCH    23
00043 #define NLE_NOCACHE         24
00044 #define NLE_BUSY            25
00045 #define NLE_PROTO_MISMATCH  26
00046 #define NLE_NOACCESS        27
00047 #define NLE_PERM            28
00048 #define NLE_PKTLOC_FILE            29
00049 #define NLE_PARSE_ERR              30
00050 #define NLE_NODEV           31
00051 #define NLE_IMMUTABLE              32
00052 #define NLE_DUMP_INTR              33
00053 
00054 #define NLE_MAX                    NLE_DUMP_INTR
```

Comment (OphirMunk) [inline src/vma/netlink/netlink_compatibility.cpp]: Correction: returned values are always negative (minus the definition)
Please ignore previous comment

**Comment by OphirMunk** [comment]
Merged #261

**Pull request closed without merging**